app/src/main/java/com/google/droidjump/models/Droid.java

```diff
@@ -34,21 +37,19 @@
     // the variable is responsible for how much time ticks must pass to change a bitmap so that animate droid smoothly
     public final static int animationStepTicks = 2;
 
-
-    public Droid(int x, int y, Resources resources) {
-        super(x, y);
-        Bitmap fullDroidPicture = BitmapFactory.decodeResource(resources, R.mipmap.droid);
+    public Droid (int x, int y, Resources resources){
+        super(x, y, R.mipmap.droid, resources);
         int droidCount = GameConstants.DROID_COUNT_ON_FULL_DROID_PICTURE;
         droidTypes = new Bitmap[droidCount];
-        int droidWidth = fullDroidPicture.getWidth() / droidCount;
-        int droidHeight = fullDroidPicture.getHeight();
+        int droidWidth = picture.getWidth() / droidCount;
+        int droidHeight = picture.getHeight();
         for (int i = 0; i < droidCount; i++) {
-            droidTypes[i] = Bitmap.createBitmap(fullDroidPicture, /* x= */ droidWidth * i,
+            droidTypes[i] = Bitmap.createBitmap(picture, /* x= */ droidWidth * i,
                     /* y= */ 0, droidWidth, droidHeight);
         }
-        setBitmap(droidTypes[GameConstants.DROID_FIRST_STEP_INDEX]);
-        this.setY(y - getBitmap().getHeight());
-        initialY = getY();
+        picture = (droidTypes[GameConstants.DROID_FIRST_STEP_INDEX]);
+        this.y = (y - picture.getHeight());
+        initialY = (int) getY();
```

Comment:
nit: why are you casting to int?

Reply:
Because previously I changed Y to float and then back again to int and forgot to change this line. I'll fix it)